### PR TITLE
Add WARNINGF, ERRORF and FATALF macros

### PR DESCRIPTION
**Pull request by GroteGnoom**
Also discussed at
https://igraph.discourse.group/t/the-unit-tests-thread/597/6

### CODE REVIEW

**Comment by szhorvat** [comment]
Thanks again! I'm happy with this but I'll leave it to @ntamas to merge.

File: include/igraph_error.h [hunk 1]
```diff
@@ -469,6 +469,36 @@ typedef enum {
...
+ *   with the error code. E.g. if the error code is \c IGRAPH_EINVAL,
+ *   its associated text (see  \ref igraph_strerror()) is "Invalid
+ *   value" and this string should explain which parameter was invalid
+ *   and maybe what was expected and what was recieved.
+ * \param igraph_errno The \a igraph error code.
+ * \param ... The additional arguments to be substituted into the
+ *   template string.
+ */
+
+#define IGRAPH_ERRORF(reason,igraph_errno, ...) \
```
Comment (szhorvat) [inline include/igraph_error.h:495]: Nitpick: can you add a space after `reason,`?

**Comment by szhorvat** [comment]
Actually, there is one thing I forgot: can you please add these new macros to `doc/error.xxml`, so that they would be included in the documentation?

File: include/igraph_error.h (hunk 1 above)
Comment (ntamas) [reply include/igraph_error.h]: Added them myself. The space was missing in two other places as well.

**Review by ntamas — APPROVED** [review]
Please extend `doc/error.xxml` with the new macros and then it's good to go.

**Comment by ntamas** [comment]
Thanks a lot!

**Pull request merged**